### PR TITLE
Issue #261 Resolved Error and Changes Needed

### PR DESCRIPTION
**Pull request by Boorclark**
Mostaphe and I created the front and back end for Issue #261. 

Changes Made: 
admin/volunteers.py - Created datePassed variable to call on bgDate. Added to set background check attribute.

logic/volunteers.py - Added parameter datePassed, Updated update variable.

volunteerProfile.js -  Updated the ajax as well as updated data to allow for bgDate. 

Also created the front end by adding a column with the date picker.
image

### CODE REVIEW

File: app/controllers/admin/volunteers.py [hunk 1]
```diff
@@ -107,5 +107,7 @@ def updateBackgroundCheck():
         user = eventData['user']
         checkPassed = int(eventData['checkPassed'])
         type = eventData['bgType']
-        setUserBackgroundCheck(user,type, checkPassed)
+        datePassed = eventData['bgDate']
+        print(datePassed)
```
Comment (BrianRamsay) [inline app/controllers/admin/volunteers.py]: Remove debug output

File: app/models/backgroundCheck.py [hunk 2]
```diff
@@ -7,6 +7,5 @@ class BackgroundCheck(baseModel):
     type = ForeignKeyField(BackgroundCheckType)
     passBackgroundCheck = BooleanField(default=False)
     datePassed = DateField(null=True)
-
```
Comment (BrianRamsay) [inline app/models/backgroundCheck.py]: There should still be a new line between the fields and the key

File: app/static/js/volunteerProfile.js [hunk 3]
```diff
@@ -80,25 +80,28 @@ $(document).ready(function(){
...
   $(".backgroundCheck").change(function () { // Updates the Background check of a volunteer in the database
+    checkType = $(this).attr("id")
     let data = {
-        checkPassed : $(this).val(), //Expected to be either a 0 or a 1
-        user: $(this).data("username"), //Expected to be the username of a volunteer in the database
-        bgType: $(this).attr("id") // Expected to be the ID of a background check in the database
+        checkPassed : $(this).val(),      // Expected to be either a 0 or a 1
+        user: $(this).data("username"),   // Expected to be the username of a volunteer in the database
+        bgType: checkType,       // Expected to be the ID of a background check in the database
+        bgDate: $("#" + checkType + "_date").val()
```
Comment (BrianRamsay) [inline app/static/js/volunteerProfile.js]: add a description like the other elements in the `data` dictionary

File: app/static/js/volunteerProfile.js [hunk 4]
```diff
@@ -80,25 +80,28 @@ $(document).ready(function(){
...
     }
+
     $.ajax({
       url: "/updateBackgroundCheck",
       type: "POST",
       data: data,
       success: function(s){
-        location.reload()
+        //location.reload()
```
Comment (BrianRamsay) [inline app/static/js/volunteerProfile.js]: Decide whether a reload is necessary. If so, uncomment this line, if not, remove it.

File: app/templates/main/volunteerProfile.html [hunk 5]
```diff
@@ -153,27 +153,29 @@ <h2>Background Check</h2>
      <thead>
        <tr>
          <th scope="col">Description</th>
+         <th scope="col"> Date Completed </th>
```
Comment (BrianRamsay) [inline app/templates/main/volunteerProfile.html]: Remove whitespace around header

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

File: app/static/js/volunteerProfile.js [hunk 6]
```diff
@@ -80,25 +80,27 @@ $(document).ready(function(){
...
   $(".backgroundCheck").change(function () { // Updates the Background check of a volunteer in the database
+    checkType = $(this).attr("id")
     let data = {
-        checkPassed : $(this).val(), //Expected to be either a 0 or a 1
-        user: $(this).data("username"), //Expected to be the username of a volunteer in the database
-        bgType: $(this).attr("id") // Expected to be the ID of a background check in the database
+        checkPassed : $(this).val(),      // Expected to be either a 0 or a 1
+        user: $(this).data("username"),   // Expected to be the username of a volunteer in the database
+        bgType: checkType,       // Expected to be the ID of a background check in the database
+        bgDate: $("#" + checkType + "_date").val()  // Expected to be the type (ex. FBI) and  _date
```
Comment (BrianRamsay) [inline app/static/js/volunteerProfile.js]: The comment should be about the date itself, not what the line is doing. like, `// The date of the background check completion in YYYY/MM/DD format` (or whatever the format is)

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

**Review by Boorclark — COMMENTED** [review]
The comment has been fixed by Mostaphe.

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**